### ISSUE TITLE
Confusion with regards to cmsCreateTransform

### ISSUE THREAD

**Issue by pathcore-tasos** [mm2/Little-CMS#132]
Hi there! 

I have a question in regards to cmsCreateTransform, with the signature below:
```C
cmsHTRANSFORM cmsCreateTransform(cmsHPROFILE Input,
                                 cmsUInt32Number InputFormat,
                                 cmsHPROFILE Output,
                                 cmsUInt32Number OutputFormat,
                                 cmsUInt32Number Intent,
                                 cmsUInt32Number dwFlags);
```
I understand the intent of this function is to create an object capable of transforming a pixel in `Input` to `Output`. However, what is the intended purpose of `InputFormat` and `OutputFormat`? In the documentation, it is written:
> Formatters are used to describe how bitmap buffers are organized.

If ICC profiles expect pixels of a particular format, say RGB8, why is it necessary to specify this information in the `InputFormat` parameter? Furthermore, how is one supposed to specify the value of `OutputParameter` if this is information contained inside the `Output` ICC profile? 

To exemplify, say I have a buffer of RGB8 pixels I would like to convert using an ICC profile somewhere on disk. It just so happens this output ICC profile is CMYK, but without querying information from the Output profile, there is no way to specify the output format correctly.

In short, what is the purpose of `InputFormat` and `OutputFormat` and how can one find correct values to use?

Thanks!

**Comment by mm2** [comment]
> I understand the intent of this function is to create an object 
capable of transforming a pixel in |Input| to |Output|

|Please see the tutorial. This function creates a color transform for 
translating bitmaps. ICC profiles does *not* expect particular formats, 
only color spaces. i.e. a profile handles CMYK but you can use 8 or 16 
bits per channel, you can use additional channels for alpha blending, or 
you can use floating point. This is not in the profile and you have to 
specify it. In the tutorial this is explained in detail.
|

|
|
